### PR TITLE
DDF-3016 Update catalog-core-solrcommands to use new Karaf command annotations

### PR DESCRIPTION
**Pull request by clockard**
#### What does this PR do?
Update catalog-core-solrcommands to use new Karaf command annotations
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@vinamartin @mcalcote @emmberk 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@figliold
@shaundmorris

#### How should this be tested? (List steps with links to updated documentation)
Verify build and solr:backup command works
#### What are the relevant tickets?
[DDF-3016](https://codice.atlassian.net/browse/DDF-3016)
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by clockard** [comment]
build now

**Comment by clockard** [comment]
build now

File: catalog/core/catalog-core-solrcommands/pom.xml [hunk 1]
```diff
@@ -198,7 +204,8 @@
                             javax.xml.xpath,
                             org.apache.commons.codec.binary,
                             org.apache.commons.logging; version="[1.1.0,2.0.0)",
-                            org.apache.felix.gogo.commands,
+                            org.apache.karaf.shell.api.action,
```
Comment (shaundmorris) [inline catalog/core/catalog-core-solrcommands/pom.xml]: ❔ Do we need to embed the karaf shell API across all commands? I assume a bundle isn't exporting those API or this is the best way to do it anyway, just confirming.

Comment (clockard) [inline catalog/core/catalog-core-solrcommands/pom.xml]: We haven't needed to embed them. I think karaf is exporting them.

**Pull request merged**